### PR TITLE
[Refactor] Prefix renaming for CommunicationUI- to CallComposite-

### PR DESCRIPTION
**Pull request by jimchou-dev**
## Purpose

* Prefix renaming for CommunicationUI- to CallComposite- for supported locale, error event, and error code

## Does this introduce a breaking change?

```
[ ] Yes
[ x ] No
```

## Pull Request Type
What kind of change does this Pull Request introduce?

```
[ ] Bugfix
[ ] Feature
[ ] Code style update (formatting, local variables)
[ x ] Refactoring (no functional changes, no api changes)
[ ] Documentation content changes
[ ] Other... Please describe:
```

## What to Check
Verify that the following are valid
* All UI and unit test passing, app remain functional

### CODE REVIEW

File: AzureCommunicationUI/sdk/AzureCommunicationUICalling/AzureCommunicationUICallingTests/Redux/Reducer/ErrorReducerTests.swift [hunk 1]
```diff
@@ -84,10 +84,10 @@ class ErrorReducerTests: XCTestCase {
     }
 
     func test_handleErrorReducer_reduce_when_statusErrorCallDeniedAndCallReset_then_returnErrorState_categoryCallState() {
-        let state = ErrorState(error: CommunicationUIErrorEvent(code: CallCompositeErrorCode.callDenied,
+        let state = ErrorState(error: CallCompositeErrorEvent(code: CallCompositeErrorCode.callDenied,
```
Comment (kevinyulu) [inline AzureCommunicationUI/sdk/AzureCommunicationUICalling/AzureCommunicationUICallingTests/Redux/Reducer/ErrorReducerTests.swift:87]: Should we add prefix to ErrorState?

**Review by kevinyulu — DISMISSED** [review]
tested, lgtm

File: AzureCommunicationUI/sdk/AzureCommunicationUICalling/AzureCommunicationUICallingTests/Redux/Reducer/ErrorReducerTests.swift (hunk 1 above)
Comment (jimchou-dev) [reply AzureCommunicationUI/sdk/AzureCommunicationUICalling/AzureCommunicationUICallingTests/Redux/Reducer/ErrorReducerTests.swift:87]: ErrorState is internal and not public facing

File: AzureCommunicationUI/AzureCommunicationUIDemoApp/AzureCommunicationUIDemoApp/EnvConfig.swift [hunk 2]
```diff
@@ -36,7 +36,7 @@ class EnvConfigSubject: ObservableObject {
 
     @Published var selectedAcsTokenType: ACSTokenType = .token
     @Published var selectedMeetingType: MeetingType = .groupCall
-    @Published var locale: Locale = CommunicationUISupportedLocale.en
+    @Published var locale: Locale = CallCompositeSupportedLocale.en
```
Comment (vhuseinova-msft) [inline AzureCommunicationUI/AzureCommunicationUIDemoApp/AzureCommunicationUIDemoApp/EnvConfig.swift:39]: Should these changes be included in CHANGELOG?

**Review by vhuseinova-msft — DISMISSED** [review]

File: AzureCommunicationUI/AzureCommunicationUIDemoApp/AzureCommunicationUIDemoApp/EnvConfig.swift (hunk 2 above)
Comment (jimchou-dev) [reply AzureCommunicationUI/AzureCommunicationUIDemoApp/AzureCommunicationUIDemoApp/EnvConfig.swift:39]: There seems to be more changes to come with the renaming, will create separate PR for all the CHANGELOG related changes

**Review by vhuseinova-msft — APPROVED** [review]

**Review by kevinyulu — APPROVED** [review]

**Pull request merged**